Packages/com.unity.inputsystem/InputSystem/InputUpdateType.cs

```diff
@@ -71,27 +71,95 @@ public enum InputUpdateType
 
     internal static class InputUpdate
     {
-        public static bool s_HaveUpdatedStepCount;
         public static InputUpdateType s_LastUpdateType;
-        public static uint s_UpdateStepCount;
+        public static UpdateStepCount s_PlayerUpdateStepCount;
+        #if UNITY_EDITOR
+        public static UpdateStepCount s_EditorUpdateStepCount;
+        public static uint s_UpdateStepCount => s_LastUpdateType == InputUpdateType.Editor ? s_EditorUpdateStepCount.value : s_PlayerUpdateStepCount.value;
+        #else
+        public static uint s_UpdateStepCount => s_PlayerUpdateStepCount.value;
+        #endif
         public static uint s_LastUpdateRetainedEventBytes;
         public static uint s_LastUpdateRetainedEventCount;
 
+        [Serializable]
+        public struct UpdateStepCount
+        {
+            public bool wasUpdated;
+            public uint value;
+
+            public void OnBeforeUpdate()
+            {
+                wasUpdated = true;
+                value++;
+            }
+
+            public void OnUpdate()
+            {
+                // only increment if OnBeforeUpdate was not called
+                if (!wasUpdated)
+                    value++;
+                wasUpdated = false;
```

Comment:
is this correct - wasUpdated being set to false, since above condition is executed when wasUpdated == false, so I would assume if you call OnUpdate again, value will not increment

Reply:
If we call OnBeforeUpdate + OnUpdate the value should only be incremented once as it's treated as one frame, but if we call OnUpdate + OnUpdate then it's two frames and value is incremented twice

Reply:
yes, that's fine, what I am talking about is the variable name, it's counter intuitive :) I read it as "was updated", so it's weird that's being set to false from function "where Update is happening"